pkg/operator/vsphereproblemdetector/vsphere_problem_detector_starter.go

```diff
@@ -2,31 +2,45 @@ package vsphereproblemdetector
 
 import (
 	"context"
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
 	"time"
 
 	configv1 "github.com/openshift/api/config/v1"
 	operatorapi "github.com/openshift/api/operator/v1"
 	openshiftv1 "github.com/openshift/client-go/config/listers/config/v1"
 	"github.com/openshift/cluster-storage-operator/assets"
 	"github.com/openshift/cluster-storage-operator/pkg/csoclients"
-	"github.com/openshift/cluster-storage-operator/pkg/operatorclient"
 	"github.com/openshift/library-go/pkg/controller/factory"
 	"github.com/openshift/library-go/pkg/controller/manager"
+	"github.com/openshift/library-go/pkg/operator/csi/csidrivercontrollerservicecontroller"
+	"github.com/openshift/library-go/pkg/operator/deploymentcontroller"
 	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
 	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehash"
 	"github.com/openshift/library-go/pkg/operator/staticresourcecontroller"
 	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	appsv1 "k8s.io/api/apps/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	coreinformers "k8s.io/client-go/informers/core/v1"
 	"k8s.io/klog/v2"
 )
 
 const (
-	infraConfigName = "cluster"
+	infraConfigName                     = "cluster"
+	vSphereProblemDetectorOperatorImage = "VSPHERE_PROBLEM_DETECTOR_OPERATOR_IMAGE"
+	secretName                          = "vsphere-cloud-credentials"
+	cloudConfigNamespace                = "openshift-config"
 )
 
 type VSphereProblemDetectorStarter struct {
 	controller     manager.ControllerManager
-	operatorClient *operatorclient.OperatorClient
+	operatorClient v1helpers.OperatorClientWithFinalizers
```

Comment:
Why do you need `OperatorClientWithFinalizers`? It doesn't seem like you're using it the extra methods.

Reply:
`DeploymentController` needs it, because it supports removable operands.